Crate polars-arrow doesn't compile with feature "arrow_rs"

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

Include the polars-arrow crate with the "arrow_rs" feature.

``` Cargo.toml
polars-arrow = { version = "0.38", features = ["arrow_rs"] }
```

### Log output

_No response_

### Issue description

   Compiling polars-arrow v0.38.3
error[E0004]: non-exhaustive patterns: `DataType::BinaryView`, `DataType::Utf8View`, `DataType::ListView(_)` and 1 more not covered
   --> /Users/<user>/.cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-arrow-0.38.3/src/datatypes/mod.rs:247:15
    |
247 |         match value {
    |               ^^^^^ patterns `DataType::BinaryView`, `DataType::Utf8View`, `DataType::ListView(_)` and 1 more not covered
    |
note: `DataType` defined here
   --> /Users/<user>/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-schema-51.0.0/src/datatype.rs:44:1
    |
44  | pub enum DataType {
    | ^^^^^^^^^^^^^^^^^
...
207 |     BinaryView,
    |     ---------- not covered
...
226 |     Utf8View,
    |     -------- not covered
...
239 |     ListView(FieldRef),
    |     -------- not covered
...
254 |     LargeListView(FieldRef),
    |     ------------- not covered
    = note: the matched value is of type `DataType`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern as shown, or multiple match arms
    |
303 ~             },
304 ~             _ => todo!(),
    |

For more information about this error, try `rustc --explain E0004`.
error: could not compile `polars-arrow` (lib) due to 1 previous error


### Expected behavior

Compile without error

### Installed versions

<details>

polars-arrow = { version = "0.38", features = ["arrow_rs"] }

</details>

--- a/crates/polars-arrow/src/datatypes/mod.rs
+++ b/crates/polars-arrow/src/datatypes/mod.rs
@@ -303,7 +303,7 @@ impl From<arrow_schema::DataType> for ArrowDataType {
             },
             // This ensures that it doesn't fail to compile when new variants are added to Arrow
             #[allow(unreachable_patterns)]
-            dtype => unimplemented!("unsupported datatype: {dtype}")
+            dtype => unimplemented!("unsupported datatype: {dtype}"),
         }
     }
 }

--- a/crates/polars-arrow/src/datatypes/mod.rs
+++ b/crates/polars-arrow/src/datatypes/mod.rs
@@ -301,6 +301,8 @@ impl From<arrow_schema::DataType> for ArrowDataType {
             DataType::RunEndEncoded(_, _) => {
                 panic!("Run-end encoding not supported by polars_arrow")
             },
+            #[allow(unreachable_patterns)]
+            _ => unimplemented!("needs arrow-schema >= v51.0.0"),
         }
     }
 }

--- a/crates/polars-arrow/src/datatypes/mod.rs
+++ b/crates/polars-arrow/src/datatypes/mod.rs
@@ -301,8 +301,9 @@ impl From<arrow_schema::DataType> for ArrowDataType {
             DataType::RunEndEncoded(_, _) => {
                 panic!("Run-end encoding not supported by polars_arrow")
             },
+            // This ensures that it doesn't fail to compile when new variants are added to Arrow
             #[allow(unreachable_patterns)]
-            _ => unimplemented!("needs arrow-schema >= v51.0.0"),
+            dtype => unimplemented!("unsupported datatype: {dtype}")
         }
     }
 }

